app/src/main/java/org/mozilla/fenix/search/SearchDialogController.kt

```diff
@@ -211,6 +215,12 @@ class SearchDialogController(
         navController.navigateSafe(R.id.searchDialogFragment, directions)
     }
 
+    override fun handleClickHistorySearch() {
+        clearToolbarFocus()
+        val directions = SearchDialogFragmentDirections.actionGlobalHistoryFragment(search = true)
```

Comment:
I think we already talked about this before, but were we landing the 'hacky' solution while working towards a true solution or were there some new developments where the 'hacky' solution is the true solution?

Reply:
This hacky solution is the cleanest solution I can find at the moment.  Tried to add in A-C side or on Fenix side as a search engine but keep running into issues that require fixes that create more problems.